app/main.py

```diff
@@ -1,3 +1,92 @@
+from __future__ import annotations
+from typing import Union
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(
+            self,
+            km: Union[int, float]
+    ) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(
+            self,
+            other: Union[int, float, Distance]
+    ) -> Distance:
+        if isinstance(other, Union[int, float]):
+            return Distance(self.km + other)
+        else:
+            return Distance(self.km + other.km)
+
+    def __iadd__(
+            self,
+            other: Union[int, float, Distance]
+    ) -> Distance:
+        if isinstance(other, Union[int, float]):
+            self.km += other
+        else:
+            self.km += other.km
+        return self
+
+    def __mul__(
+            self,
+            other: Union[int, float]
+    ) -> Distance:
+        return Distance(self.km * other)
+
+    def __truediv__(
+            self,
+            other: Union[int, float]
+    ) -> Distance:
+        return Distance(round((self.km / other), 2))
```

Comment:
The `__truediv__` method should handle cases where `other` is a `Distance` object. Additionally, you should include a check to prevent division by zero to avoid runtime errors.